<launch>
  <!-- Start map server -->
  <arg name="map_file" default="$(find multi_map_nav)/maps/map1.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

  <!-- AMCL localization -->
  <node name="amcl" pkg="amcl" type="amcl" output="screen">
    <param name="use_map_topic" value="true"/>
    <param name="odom_frame_id" value="odom"/>
  </node>

  <!-- Move base -->
  <node name="move_base" pkg="move_base" type="move_base" output="screen">
    <rosparam file="$(find multi_map_nav)/config/move_base_params.yaml" command="load"/>
  </node>

  <!-- Your custom multi-map navigation server -->
  <node pkg="multi_map_nav" type="navigation_server" name="navigation_server" output="screen">
    <param name="wormhole_db_path" value="$(find multi_map_nav)/database/wormholes.db" />
    <param name="map_folder" value="$(find multi_map_nav)/maps" />
  </node>

  <!-- Launch RViz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find multi_map_nav)/config/nav.rviz" required="true" />
</launch>
